pkg/distributor/otel.go

```diff
@@ -48,14 +47,13 @@ func OTLPHandler(
 	limits *validation.Overrides,
 	reg prometheus.Registerer,
 	push PushFunc,
+	logger log.Logger,
 ) http.Handler {
 	discardedDueToOtelParseError := validation.DiscardedSamplesCounter(reg, otelParseError)
 
-	return handler(maxRecvMsgSize, sourceIPs, allowSkipLabelNameValidation, limits, push, func(ctx context.Context, r *http.Request, maxRecvMsgSize int, dst []byte, req *mimirpb.PreallocWriteRequest) ([]byte, error) {
+	return handler(maxRecvMsgSize, sourceIPs, allowSkipLabelNameValidation, limits, push, logger, func(ctx context.Context, r *http.Request, maxRecvMsgSize int, dst []byte, req *mimirpb.PreallocWriteRequest, logger log.Logger) ([]byte, error) {
 		var decoderFunc func(buf []byte) (pmetricotlp.ExportRequest, error)
 
-		logger := log.WithContext(ctx, log.Logger)
```

Comment:
I think we can leave the log.WithContext, it adds traceid as part of logger, which spanlogger.NewWithLogger doesn't seems to do it.

Reply:
@ying-jeanne but `handler` calls this function (as `parser`) with a logger that's [instantiated with context](https://github.com/grafana/mimir/pull/6652/files/342bda5594704b9b8a19a2fe057585705fb053b2#diff-8a770bd239c72b7de7bc51e232145f906f94d15727124a15fbe0d46c2d98def7R90). Do you agree that we no longer need the `log.WithContext` call in this function?